modules/effects/src/provide_effects.ts

```diff
@@ -0,0 +1,64 @@
+import { ENVIRONMENT_INITIALIZER, inject, Provider, Type } from '@angular/core';
+import { Store } from '@ngrx/store';
+import { EffectsRunner } from './effects_runner';
+import { EffectSources } from './effect_sources';
+import { rootEffectsInit as effectsInit } from './effects_actions';
+
+/**
+ * Runs provided effects. Can be called at the root or feature level.
+ * Unlike the `StoreModule.forRoot` method, this function does not need
+ * to be called at the root level if there are no root effects.
+ *
+ * @usageNotes
+ *
+ * ### Providing root effects
+ *
+ * ```ts
+ * bootstrapApplication(AppComponent, {
+ *   providers: [provideEffects([RouterEffects])],
+ * });
+ * ```
+ *
+ * ### Providing feature effects
+ *
+ * ```ts
+ * const booksRoutes: Route[] = [
+ *   {
+ *     path: '',
+ *     providers: [provideEffects([BooksApiEffects])],
+ *     children: [
+ *       { path: '', component: BookListComponent },
+ *       { path: ':id', component: BookDetailsComponent },
+ *     ],
+ *   },
+ * ];
+ * ```
+ */
+export function provideEffects(effects: Type<unknown>[]): Provider[] {
+  return [
+    effects,
+    {
```

Comment:
We also need to add a dependency on the Store/StoreFeature to be injected so we know its loaded. We do this with the StoreRootModule and StoreFeatureModule being injected into the EffectsRootModule and EffectsFeatureModule currently.